applications/boltz2_fixed_affinity_screen/template.yaml

```diff
@@ -196,33 +193,46 @@ jobs:
     env:
       - HOME=/data/home/wresch
       - BOLTZ_CACHE={{ $boltz_cache }}
-      - PYTHONWARNINGS="ignore::FutureWarning"
-    command:
-      - /bin/bash
-      - "-c"
-      - |
-        lockfile="${BOLTZ_CACHE}.lock"
-        while :; do
-          if mkdir "${lockfile}" 2>/dev/null; then
-            # We got the lock
-            if [[ ! -d "${BOLTZ_CACHE}" ]]; then
-              printf "Downloading model to %s\n" "${BOLTZ_CACHE}"
-              mkdir -p "${BOLTZ_CACHE}"
-              python -c "from pathlib import Path; from boltz.main import download_boltz2; download_boltz2(Path('${BOLTZ_CACHE}'))" || exit 1
-            fi
-            rmdir "${lockfile}"
-            break
-          else
-            # Wait for the lock to be released
-            printf "Waiting for model cache download to complete...\n"
-            sleep 1
+      - PYTHONWARNINGS=ignore::FutureWarning
+    script: |
+      #!/bin/bash
+      PATH=/app/env/bin:${PATH}
+      if [[ ! -x /app/env/bin/python ]]; then
+        echo "Error: Python not found at /app/env/bin/python"
+        exit 1
+      fi
+
+      lock="${BOLTZ_CACHE}.lock"
+      mkdir -p "$(dirname "${lock}")"
+
+      cleanup_lock() {
+        rmdir "${lock}" 2>/dev/null || true
+      }
+
+      n=0
+      while :; do
+        if mkdir "${lock}" 2>/dev/null; then
+          trap cleanup_lock EXIT
+          # We got the lock
+          if [[ ! -d "${BOLTZ_CACHE}" ]]; then
+            printf "Downloading model to %s\n" "${BOLTZ_CACHE}"
+            mkdir -p "${BOLTZ_CACHE}"
+            python -c "from pathlib import Path; from boltz.main import download_boltz2; download_boltz2(Path('${BOLTZ_CACHE}'))" || exit 1
           fi
-        done
-        mkdir -p outputs
+          rmdir "${lock}"
+          trap - EXIT
```

Comment:
Are both of these lines needed to make sure that the lock is removed or is the trap enough?